cfme/v2v/migration_plans.py

```diff
@@ -0,0 +1,636 @@
+                self.migration_request_details_list.is_in_progress(vm)
+                and (clock_reading1 < clock_reading2)
+            )
+        return not any(migration_plan_in_progress_tracker)
+
+
+@attr.s
+class MigrationPlan(BaseEntity):
+    """Class representing v2v Migration Plan"""
+
+    name = attr.ib()
+    infra_map = attr.ib()
+    vm_list = attr.ib()
+    description = attr.ib(default=None)
+    csv_import = attr.ib(default=False)
+    target_provider = attr.ib(default=None)
+    osp_security_group = attr.ib(default=None)
+    osp_flavor = attr.ib(default=None)
+    pre_playbook = attr.ib(default=None)
+    post_playbook = attr.ib(default=None)
+
+    def is_plan_in_progress(self):
+        """MIQ V2V UI is going through redesign as OSP will be integrated."""
+        view = navigate_to(self, "InProgress")
+
+        def _in_progress():
+            try:
+                is_plan_visible = view.progress_card.is_plan_visible(self.name)
+                if is_plan_visible:
+                    try:
+                        plan_time_elapsed = view.progress_card.get_clock(self.name)
+                        new_msg = "time elapsed for migration: {time}".format(
+                            time=plan_time_elapsed
+                        )
+                    except NoSuchElementException:
+                        new_msg = "playbook is executing.."
+                        pass
+                    logger.info(
+                        "For plan {plan_name}, is plan in progress: {visibility}, {message}".format(
+                            plan_name=self.name, visibility=is_plan_visible, message=new_msg
+                        )
+                    )
+                return not is_plan_visible
+            except ItemNotFound:
+                return True
+
+        return wait_for(
+            func=_in_progress,
+            message="migration plan is in progress, be patient please",
+            delay=5,
+            num_sec=1800,
+        )
+
+    def is_migration_complete(self):
+        """Uses search box to find migration plan, return True if found.
+        Args:
+            migration_plan name: (object) Migration Plan name
+        """
+        view = navigate_to(self, "Complete")
+        view.wait_displayed()
+        view.items_on_page.item_select("15")
+        view.search_box.fill("{}\n\n".format(self.name))
+        return (self.name in view.plans_completed_list.read() and
+                view.plans_completed_list.is_plan_succeeded(self.name))
+
+    def migration_plan_request(self):
+        view = navigate_to(self, "InProgress")
+        view.progress_card.select_plan(self.name)
+        request_view = self.create_view(MigrationPlanRequestDetailsView, wait="10s")
+        request_details_list = request_view.migration_request_details_list
+        view.items_on_page.item_select("15")
+        self.is_plan_in_progress()
+        return request_details_list
```

Comment:
What's the intent/purpose of this method? I find it a bit odd that it is returning a widget object. This almost seems like a navigation destination?   It seems like `self.is_plan_in_progress()` is doing _something_ to the view that we just created, but its not clear what?

Reply:
Fixed